doc/dev/docstring_typehint.md

```diff
@@ -0,0 +1,185 @@
+            object to return a :class:`~azure.ai.formrecognizer.CustomFormModel`.
+        :rtype: ~azure.core.polling.LROPoller[~azure.ai.formrecognizer.CustomFormModel]
+        :raises ~azure.core.exceptions.HttpResponseError:
+            Note that if the training fails, the exception is raised, but a model with an
+            "invalid" status is still created. You can delete this model by calling :func:`~delete_model()`
+
+        .. versionadded:: v2.1
+            The *model_name* keyword argument
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_train_model_without_labels.py
+                :start-after: [START training]
+                :end-before: [END training]
+                :language: python
+                :dedent: 8
+                :caption: Training a model (without labels) with your custom forms.
+        """
+        ...
+```
+
+The first portion of this docstring is a general description of what the method does. Following the general description of the method is a **required new line** and then documentation for each of the parameters, optional keyword arguments, returned objects, and potentially raised errors.
+
+Required parameters can be documented in one-line or two-lines. Both options can be used in a single docstring for documenting different parameters without issue.
+1. This option works best for parameters that are one of the basic types (`str`, `int`, `bool`, `bytes`, `float`, etc.)
+```python
+:param <type> <param_name>: <Description of the parameter>
+```
+2. This option works best for parameters that use custom models or have many different options
+```python
+:param <param_name>: <Description of the parameter>
+:type <param_name>: <param_type>
+```
+
+Optional keyword arguments can in one-line or two-lines. Both options can be used in a single docstring for documenting different keywords without issue.
+1. This option works best for keyword args that are one of the basic types (`str`, `int`, `bool`, `bytes`, `float`, etc.)
+```python
+:keyword <type> <keyword_name>: <Description of the keyword>
+```
+2. This option works best for keyword that use custom models or have many different options
+```python
+:keyword <keyword_name>: <Description of the keyword>
+:paramtype <keyword_name>: <keyword_type>
+```
+
+The returned object is documented on two lines, the first describing what the returned object is and the second describing what the returned type is.
+```python
+"""
+:return: <Description of the returned object>
+:rtype: <Type of the returned object>
+"""
+```
+
+Finally, describe the possible errors raised by a method:
+```python
+:raises <error1>, <error2>, or <error3>: <description>
+```
+
+All of the above information needs to be added for each public method and there **needs to be a newline after the above docstrings**. There are additional options for including examples and version specific additions.
```

Comment:
for additional options, not sure if you want to link out to this: https://review.docs.microsoft.com/en-us/help/onboard/admin/reference/python/documenting-api?branch=master#supported-sphinx-directives